src/bidfactory.js

```diff
@@ -15,15 +15,15 @@ var utils = require('./utils.js');
  priceKeyString;
  */
 function Bid(statusCode, bidRequest) {
-  var _bidId = (bidRequest && bidRequest.bidId) || utils.getUniqueIdentifierStr();
   var _bidSrc = (bidRequest && bidRequest.src) || 'client';
   var _statusCode = statusCode || 0;
 
   this.bidderCode = (bidRequest && bidRequest.bidder) || '';
   this.width = 0;
   this.height = 0;
   this.statusMessage = _getStatus();
-  this.adId = _bidId;
+  this.adId = utils.getUniqueIdentifierStr();
+  this.requestId = bidRequest && bidRequest.bidId;
```

Comment:
I noticed that the currency file calls [bidfactory.createBid()](https://github.com/prebid/Prebid.js/blob/master/modules/currency.js#L223-L226) when conversions fail.  There might be some (minor) repercussions since the `adId` will no longer be maintained in the new bid.  Might want to at least change [this line in the currency module](https://github.com/prebid/Prebid.js/blob/master/modules/currency.js#L225), from `bidId: bid.adId` to `bidId: bid.requestId` to maintain the requestId - though that isn't maintained currently.

Reply:
Thanks for the feedback regarding the `currency` file.  I pushed an update for that reference.